lib/dialects/mysql/query.js

```diff
@@ -110,6 +110,8 @@ Query.prototype.formatResults = function(data) {
     result = data[0].version;
   } else if (this.isForeignKeysQuery()) {
     result = data;
+  } else if (this.isInsertQuery() || this.isUpdateQuery()) {
+    result = [result, data.affectedRows];
```

Comment:
Not sure what your question is, but yes, I think it would always return 1 here because bulkInsert is handled differently.